### PR TITLE
68 refactor pile properties in line with pilecore api v3 definition

### LINKED ISSUE #68: Refactor pile_properties in line with pilecore-API v3 definition
- [ ] Add `PileProperties` object
  - [ ] Can be created with the new pile arguments, equal to API v3 pile_properties schema
  - [ ] Should have a `plot` method that visualizes the pile
  - [ ] Should have a `__repr__` method that displays the pile properties
  - [ ] Can be created from input, but also from the returned calculation output
- [ ] `create_multi_cpt_payload` should accept this `PileProperties` object as an argument instead of legacy pile arguments
- [ ] The API payloads should be updated to match the API v3 schema
- [ ] Update example notebook
  - [ ] Add new pile-type images
  - [ ] Add the new `PileProperties` workflow

### PR DESCRIPTION
**Pull request by tlukkezen**
closes #68

### CODE REVIEW

File: src/pypilecore/common/piles/geometry/components/round.py [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+            height = (
+                max(component_tip_level_nap, bottom_boundary_nap)
+                - component_head_level_nap
+            )
+
+            axes.add_patch(
+                patches.Rectangle(
+                    (self.cross_section_bounds[0], z_offset),
+                    self.diameter,
+                    height,
```
Comment (RDWimmers) [inline src/pypilecore/common/piles/geometry/components/round.py:426]: if a `'primary_dimension': {'length': float}` is given can we add that to the plot as well.

For example when `height_base=1`. 
image

**Review by RDWimmers — APPROVED** [review]
Maybe we can give an overview of what the different dimensions mean. We created a already something like that, right?


Also need to update the docs -> https://github.com/cemsbv/py-pilecore/blob/b4e5563d4b7fbe915ccf25acbc799547f019410a/docs/tree/reference.rst?plain=1#L57

File: src/pypilecore/common/piles/geometry/components/round.py (hunk 1 above)
Comment (tlukkezen) [reply src/pypilecore/common/piles/geometry/components/round.py:426]: fixed

File: pyproject.toml [hunk 2]
```diff
@@ -81,7 +81,7 @@ disallow_untyped_calls = true
 warn_return_any = false
 no_implicit_reexport = true
 strict_equality = true
-install_types = true
+ignore_missing_imports = true
```
[the remark this answers is not in the document]
Comment (tlukkezen) [reply pyproject.toml]: docs are updated, mypy settings fixed, image can be fixed in another PR

**Comment by RDWimmers** [comment]
image

**Review by RDWimmers — APPROVED** [review]

**Comment by tlukkezen** [comment]
# Draft Release notes:


This release connects py-pilecore to the freshly release PileCore-API v3.0.
### A new PileProperties object
One of the main breaking changes in the API is the definition of a foundation pile object. This is also reflected in py-pilecore by adding a brand new `PileProperties` object, which holds the `geometry` and `pile_type` information. This `PileProperties` object can be created with the function `create_basic_pile()`:
```diff
    create_basic_pile(
        pile_name=pile_name,
+        main_type=main_type,
        specification=specification,
        installation=installation,
        pile_shape=pile_shape,
        height_base=height_base,
+        core_secondary_dimension=core_secondary_dimension,
+        core_tertiary_dimension=core_tertiary_dimension,
+        base_secondary_dimension=base_secondary_dimension,
+        base_tertiary_dimension=base_tertiary_dimension,
+        core_diameter=core_diameter,
+        base_diameter=base_diameter,
+        pile_material=pile_material,
+        custom_material=custom_material,
        settlement_curve=settlement_curve,
        adhesion=adhesion,
        alpha_p=alpha_p,
        alpha_s_clay=alpha_s_clay,
        alpha_s_sand=alpha_s_sand,
        beta_p=beta_p,
        pile_tip_factor_s=pile_tip_factor_s,
        is_auger=is_auger,
        is_low_vibrating=is_low_vibrating,
        negative_fr_delta_factor=negative_fr_delta_factor,
    )
```
Most of the input arguments were already known as arguments for creating the API payload, but there are also some new arguments, as marked above.
The changes in the pile dimensions are probably standing out the most. They reflect the changes in the pile geometry model in the API, which is made to be more modular. In the new model, we always require a *core* component of the pile, and allow extra components to be formed around it (like a *Russian doll* model). Therefore we now require the dimensions of the *core*, while the *base* dimensions are optional.
It can also be seen that the pile can also be assigned a *material*, which sets properties like the *E-modulus* and the *color* for plotting purposes.
The pile and its separete components can be visualized with plotting methods, so that you can inspect the pile shape in `py-pilecore` before sending it to the API.
Furthermore it is important to notice that we also renamed and simplified the definition of the `default_pile` (previously `pile_type_specification`). Not just the name is shorter: We also changed the required input for creating the pile-type, so be sure to check out the new pile_type tables!

**Pull request merged**